### DISCUSSION TITLE
Error in PriceActionProcessor

### DISCUSSION THREAD

**Discussion by twin-elements** [coreshop/CoreShop#2759] [Q&A] [answered]
Hi.
I found an error in the CoreShop\Component\PriceActionProcessor.

In the parameter $configutation obtains data in the format
![image](https://github.com/user-attachments/assets/856604a2-64aa-48d9-a2f1-bbd67e5fd892)

Price is string, but $this->moneyConverter->convert($price,...) requires price to be int

Because of this, we get an error
```
An exception has been thrown during the rendering of a template ("CoreShop\Component\Currency\Converter\CurrencyConverter::convert(): Argument #1 ($value) must be of type int, string given, called in /var/www/html/vendor/coreshop/core-shop/src/CoreShop/Component/Product/Rule/Action/PriceActionProcessor.php on line 55").
```
in my opinion, they should convert the price to integer
https://github.com/coreshop/CoreShop/blob/e94b1024abd8176d9de162d578e8029da28b5d9c/src/CoreShop/Component/Product/Rule/Action/PriceActionProcessor.php#L58
![image](https://github.com/user-attachments/assets/a561f2bb-a394-4470-8f8e-e1af27504fbf)

I can, of course, implement the fix

**Comment by dpfaffenbauer** [comment]
@twin-elements yes please

**Reply by twin-elements** [reply, marked as answer]
[https://github.com/coreshop/CoreShop/pull/2760](url)  done
